BadHabbit/BadHabbit/Kit/DesignKit/DropDownViewPresenter.swift

```diff
@@ -0,0 +1,175 @@
+            )
+        )
+        
+        if animated {
+            let initialFrame = CGRect(
+                x: targetFrame.minX,
+                y: targetFrame.minY,
+                width: targetFrame.width,
+                height: 0
+            )
+            
+            dropDownView.frame = initialFrame
+            overlay.backgroundColor = UIColor.clear
+            
+            UIView.animate(withDuration: 0.3, delay: 0.0, options: [.curveEaseOut]) {
+                dropDownView.frame = targetFrame
+                self.overlay.backgroundColor = UIColor.black.withAlphaComponent(0.2)
+            }
+        } else {
+            dropDownView.frame = targetFrame
+            overlay.backgroundColor = UIColor.black.withAlphaComponent(0.2)
+        }
+    }
+    
+    func hideDropDown(animated: Bool = true) {
+        guard let dropDownView else {
+            return
+        }
+        
+        let resetViews = {
+            self.overlay.subviews.forEach { view in view.removeFromSuperview() }
+            self.overlay.removeFromSuperview()
+        }
+        
+        if animated {
+            UIView.animate(
+                withDuration: 0.3,
+                delay: 0.0,
+                options: [.curveEaseIn],
+                animations: {
+                    dropDownView.frame = CGRect(
+                        origin: dropDownView.frame.origin,
+                        size: CGSize(width: dropDownView.frame.width, height: 0.0)
+                    )
+                    self.overlay.backgroundColor = UIColor.clear
+                },
+                completion: { _ in resetViews() }
+            )
+        } else {
+            resetViews()
+        }
+    }
+    
+    private func calculateDropDownFrame(viewFrame: CGRect, windowFrame: CGRect, preferredSize: CGSize) -> CGRect {
+        let width: CGFloat
+        let height: CGFloat
+        let x: CGFloat
+        let y: CGFloat
+        
+        width = min(windowFrame.width, preferredSize.width)
+        
+        let maxAvailableHeight = windowFrame.height - (viewFrame.maxY + dropDownOffset)
+        height = min(maxAvailableHeight, preferredSize.height)
+        
+        if viewFrame.midX - width / 2 <= 0 {
+            x = 0
+        } else if viewFrame.midX + width / 2 >= windowFrame.maxX {
+            x = windowFrame.maxX - width
+        } else {
+            x = viewFrame.midX - width / 2
+        }
+        
+        y = viewFrame.maxY + dropDownOffset
+        
+        return .init(x: x, y: y, width: width, height: height)
+    }
+    
+    @objc
+    private func overlayTapped() {
+        hideDropDown()
+    }
+}
+
+extension UIApplication {
+    var activeWindow: UIWindow? {
+        let scene = connectedScenes.first {
+            $0.activationState == .foregroundActive && $0 is UIWindowScene
+        }
+        let windowScene = scene as? UIWindowScene
+        let window = windowScene?.windows.first(where: \.isKeyWindow)
```

Comment:
I can't find what does it mean: \. in `(where: \.isKeyWindow)`

Reply:
This is the equivalent of `.first(where { $0.isKeyWindow }`. There is a function overload which accept keyPath of type.